modules/percolator/src/main/java/org/elasticsearch/percolator/PercolateQuery.java

```diff
@@ -206,14 +228,39 @@ public float score() throws IOException {
                         }
                     };
                 } else {
+                    Scorer scorer = verifiedQueriesQueryWeight.scorer(leafReaderContext);
                     return new BaseScorer(this, approximation, queries, percolatorIndexSearcher) {
 
+                        boolean notExhausted = scorer != null;
+                        DocIdSetIterator iterator = notExhausted ? scorer.iterator() : null;
```

Comment:
or maybe even better: I think you can simlify the logic below by using Lucene.asSequentialAccessBits (that we also use in function_score and filter(s) aggs) to check whether a doc matches the scorer.